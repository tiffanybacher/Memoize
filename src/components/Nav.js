import React, { Component } from 'react';
import Dropdown from '../components/Dropdown';

class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      slide: 0,
      showDropdown: false
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll)
  }

  handleScroll = () => {
    const currentScrollY = window.scrollY;

    if (currentScrollY > 220) {
      this.setState({ slide: '-80px' });
    } else {
      this.setState({ slide: '0px' });
    }
  }

  toggleDropdown = () => {
    this.setState({
      showDropdown: !this.state.showDropdown
    });
  }

  hideDropdown = () => {
    this.setState({
      showDropdown: false
    })
  }

  render() {
    if (this.state.showDropdown) {
      var dropdown = <Dropdown 
        showDropdown={this.state.showDropdown}
        hideDropdown={this.hideDropdown}
        hideMainFlashcard={this.props.hideMainFlashcard}
        showAllFlashcards={this.props.showAllFlashcards}
        showMissedFlashcards={this.props.showMissedFlashcards} />
    }

    return (
      <nav 
        className="Nav"
        style={{
          transform: `translate(0, ${this.state.slide})`,
          transition: `transform 200ms linear` 
        }} >
        <h1 className="Nav-header">Memoize</h1>
        <button className={this.state.showDropdown 
          ? "bars-btn bars-active"
          : "bars-btn"}>
          <i 
            className="fas fa-bars"
            onClick={this.toggleDropdown}>
          </i>
        </button>
        {dropdown}
      </nav>
    );
  }
}

export default Nav;