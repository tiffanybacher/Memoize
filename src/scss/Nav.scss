.Nav {
  align-items: center;
  background-color: $pink;
  color: white;
  display: flex;
  justify-content: space-between;
  position: fixed;
  top: 0;
  width: 100%;
  min-width: 320px;
  z-index: 4;
}

.Nav-header {
  font-family: "Roboto Slab";
  font-size: 2em;
  margin: 0;
  padding: 15px 25px;
}

.fa-bars {
  font-size: 2em;
  padding: 0px 25px;
}