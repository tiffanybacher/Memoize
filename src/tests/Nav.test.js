import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Nav from '../components/Nav';

const defaultState = {
  slide: 0,
  showDropdown: false
}

describe('Nav', () => {
  let wrapper; 
  
  beforeEach(() => {
    wrapper = shallow(
      <Nav />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
});