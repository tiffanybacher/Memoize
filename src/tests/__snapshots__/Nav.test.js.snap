// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<nav
  className="Nav"
  style={
    Object {
      "transform": "translate(0, 0)",
      "transition": "transform 200ms linear",
    }
  }
>
  <h1
    className="Nav-header"
  >
    Memoize
  </h1>
  <i
    className="fas fa-bars"
  />
</nav>
`;
